topo/node/ceos/ceos.go

```diff
@@ -152,10 +162,26 @@ func (n *Node) CreateCRD(ctx context.Context) error {
 		}
 		device.Spec.IntfMapping[k] = v.GetName()
 	}
+	if vendorData := config.GetVendorData(); vendorData != nil {
+		ceosLabConfig := &cpb.CEosLabConfig{}
+		if err := vendorData.UnmarshalTo(ceosLabConfig); err != nil {
+			return err
+		}
+		if toggleOverrides := ceosLabConfig.GetToggleOverrides(); toggleOverrides != nil {
+			device.Spec.ToggleOverrides = toggleOverrides
+		}
+		if waitForAgents := ceosLabConfig.GetWaitForAgents(); waitForAgents != nil {
+			device.Spec.WaitForAgents = waitForAgents
+		}
+	}
 	// Post to k8s
-	client, err := ceosclient.NewForConfig(n.RestConfig)
-	if err != nil {
-		return err
+	var err error
+	// The client can be set by a test, for example, to a fake
+	if client == nil {
```

Comment:
i would recommend 

var (
 newClient = ceosclient.NewForConfig
)

then in your test you just replace with the fake.New...

then here you don't need the nil check you just call newClient()

Reply:
hadn't thought of that, cool.